### DISCUSSION TITLE
ESLint Monthly

### DISCUSSION THREAD

**Discussion by Jay-Karia** [eslint/eslint#18828] [Ideas]
### Sections

- Version Updates & Key Changes
- Rule Insights
- Featured Plugin
- Quick Config
- Recommended Read
- Essential Tool
- Pro Tips

---

### Content Format / Layout

Version Updates and Key Changes:
- Version    
-  A brief summary of about 2-3 sentences
-  List of notable changes
- Full release link

Rule Insights:
- Rule Name
- A brief summary of the rule
- Short code example
- Link to documentation

Featured Plugin:
- Plugin Name and author
- A brief summary
- A simple getting started example
- Link to the official page

Quick Config:
- Why and when to use the config
- Config code block

Recommended Read:
- Blog post title
- Summary (TLDR)
- Link to full blog post

Essential Tool:
- Tool name
- Description
- Why and when to use
- How to use
- Link to tool page for more info

Pro Tips:
- Short description of the tip
- Code example
- Explanation
    

---

**Frequency**: 1 time a month (First day of the month)
**Tone**: Friendly
**Diagram**: https://excalidraw.com/#room=f4645330a854924d6fbf,t0Wu1Hm_nCCavi1WOB8OvQ
**Sample**: https://wry-record-de7.notion.site/Dummy-Content-108d3a4eacd28067acc4d56be55f8d03

**Comment by nzakas** [comment]
I like the idea, but we'd need someone to volunteer to write this. I know I don't have the time to commit to a regular newsletter.

Any volunteers?

**Reply by amareshsm** [reply]
I can help with this. How should we go about implementing the newsletter service? Should we use a third-party service to handle sending and managing email campaigns?

**Reply by Jay-Karia** [reply]
We can use email marketing software, such as [Brevo](https://www.brevo.com/) , [HubSpot](https://www.hubspot.com/?hubs_content=www.hubspot.com%2Fproducts%2Fcrm&hubs_content-cta=global-nav-logo-wrapper) ...

**Reply by nzakas** [reply]
Yeah, we should use some mailing list software, preferably one with a generous free tier. :smile:

We still need to figure out details:

* How frequently would messages be sent out?
* What would be included?
* What is the overall format?

**Reply by Jay-Karia** [reply]
Max 2 times a week (Monday and Thursday), this is ideal time which I can think of. But I will leave it to the one who is volunteering this.

We can add the content from "News and update" section from the website, whenever a new version is released.
Along with tips and tricks with a plugin recommendation or config.

This has a lot of potential which would help users after adding more content to it, but it can loose its purpose to serve as newsletter. (We can change the name 🙃)

Not thought about the format, but will share it soon after it's ready.

**Reply by Jay-Karia** [reply]
Another name which I thought: "ESLint Weekly", as this would not restrict us to only send news and updates but rather some additional content which would help users.

**Reply by nzakas** [reply]
@Jay-Karia is this something you're interested in working on?

My concern is that this is a lot of work. I run a once-monthly newsletter and it takes a significant amount of time to put together. I'm also not sure that there's a lot going on week over week. I'd leave it to whoever is running for the final decision, but if we're going to start a newsletter, it needs to live up to the quality of the ESLint project as a whole and not just disappear after a few weeks.

**Reply by Jay-Karia** [reply]
Yes, I would love to work on this. 🚀
I assure you the quality of the newsletter will not be compromised.

However, I will need some time to think and plan about this.
I will share some ideas and workings here before implementing.

**Reply by nzakas** [reply]
Thanks! Yeah, I want to be sure we're on the same page about what would be in such a newsletter.

**Reply by Jay-Karia** [reply]
A rough plan (https://excalidraw.com/#json=HoFS_JKUz-ONDtHsHVN85,VE_G3Hh1MdZs7W35PkdT9g)
Might change in future.

Would like to hear your thoughts.

**Reply by nzakas** [reply]
Can you please post on this discussion so we can more easily comment?

**Comment by Jay-Karia** [comment]
I am using [Brevo](https://www.brevo.com/)
It has a free tier for 300 emails / day.

https://www.brevo.com/pricing/

**Comment by Jay-Karia** [comment]


**Reply by nzakas** [reply]
Please just post the details in this discussion. I don't like needing to jump back and forth between apps.

**Reply by Jay-Karia** [reply]
Ooh, got it 😶

**Reply by Jay-Karia** [reply]
Posted the details in the discussion.

**Comment by Jay-Karia** [comment]
@nzakas Would you like to join the org created in Brevo ?

**Reply by nzakas** [reply]
I'm not sure if Brevo is where we should be doing this. 300 emails a day is not a lot. Let's focus instead on what the newsletter should be. I can see if I can get a mailing list software to sponsor us when we're ready.

**Comment by Jay-Karia** [comment]
@amareshsm Would you like to contribute here ?

**Reply by amareshsm** [reply]
Yes, I can contribute as well.

**Comment by nzakas** [comment]
My initial thought is that the format looks good, but seems like way too much to be doing on a weekly basis. Have you looked at some other open source "weekly" newsletters to see what they do?

**Reply by Jay-Karia** [reply]
I have subscribed to a JS Newsletter (2 times a week), where they write about a main topic, a JS Quiz, list of other related articles, and sponsor.

Maybe we can do it a once a month newsletter.

**Reply by nzakas** [reply]
Yeah, I think the difference with a general JS newsletter is that ESLint has a lot less going on in each week.

I do think the format would work well for a monthly newsletter, but if you want to do it weekly, then I'd say you limit to something like this:

1. Notable commits that landed OR notable commits in the release
2. Featured rule
3. Community news - news about other plugin
4. Links to blog posts, videos, etc. related to ESLint

**Reply by Jay-Karia** [reply]
Let's go with a monthly newsletter.

Will think about a new name.

**Comment by Jay-Karia** [comment]
Added the section names for the newsletter,

@nzakas What should be the ideal tone of the newsletter,
and accordingly we can update the section names and can think about the overall content

**Reply by nzakas** [reply]
Let's look at some examples from other frameworks/tools that have regular newsletters. I'm guessing the tone is friendly, fun, and informative.

**Reply by Jay-Karia** [reply]
React - https://reactnewsletter.com/ (example: https://reactnewsletter.com/issues/411)
Vue - https://news.vuejs.org/ (example: https://news.vuejs.org/issues/193)
Node - https://nodeweekly.com/ (example: https://nodeweekly.com/issues/549)

**Comment by Jay-Karia** [comment]
@nzakas,
Created a Notion page where can plan out the content of the newsletter, and manage.
If you feel this is fine, then you can share your email to join the page

Will unpublish the site after we start writing the newsletter.
Markdown in Notion is not quite good, will have to think another solution to host our newsletters (some features of md is not supported, so I have added raw md in the notion page)

Should we host on GitHub under a new repo: eslint/newsletter ?

**Reply by nzakas** [reply]
🤔 I suppose we could do these as posts on eslint.org and then just set up an email list that reads the RSS feed and sends out emails when there's a new post?

**Comment by Jay-Karia** [comment]
Sample Content:

## 📦 Version Updates

### v9.10.0

Starting with this release, ESLint now publishes its own type definitions, taking over from `@types/eslint` and to ensure timely updates. We've copied the existing types for a smooth transition and will improve them as part of our regular releases.
###### Other notable changes
- The `require-unicode-regexp` rule can now enforce usage of the `v` flag with regular expressions. Check out the new [option `requireFlag`](https://eslint.org/docs/latest/rules/require-unicode-regexp#options).
- This release also fixes problems found in the [`id-length`](https://eslint.org/docs/latest/rules/id-length) and [`no-invalid-regexp`](https://eslint.org/docs/latest/rules/no-invalid-regexp) rules.

See the [full release](https://github.com/eslint/eslint/releases/tag/v9.10.0)

### v9.11.0

The `no-useless-constructor` rule now offers suggestions for certain reported issues. This helps developers resolve problems more efficiently by providing actionable fixes.

See the [full release](https://github.com/eslint/eslint/releases/tag/v9.11.0)

## ⛔ Rule Insights

### `camelcase`

Let’s dive into the **camelcase** rule: it helps ensure your variables and function names are stylishly written in camelCase, making your code look neat and professional!
###### Usage
```js
// eslint.config.js
export default [
    {
        rules: {
	        camelcase: "always"
        }
    }
];
```

###### Invalid
```js
function func_name() {
	// ...
}
```

###### Valid
```js
function funcName() {
	// ...
}
```

For detailed info, see [documentation](https://eslint.org/docs/latest/rules/camelcase)

## 🔌 Featured Plugin

### `eslint-plugin-react`

Elevate your React development with **eslint-plugin-react**, a must-have plugin that ensures your components follow best practices and keeps your codebase clean and efficient!
###### Installation
```shell
npm install eslint eslint-plugin-react --save-dev
```

###### Quick Start
```js
// eslint.config.js
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ]
```

###### Key Rules
1. `no-namespace` - Enforce that namespaces are not used in React elements ([more info](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-namespace.md))
2. `jsx-uses-vars` - Disallow variables used in JSX to be incorrectly marked as unused ([more info](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md))
3. `jsx-key` - Disallow missing "key" props in iterators/collection literals ([more info](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-key.md))

For more details, see [GitHub](https://github.com/jsx-eslint/eslint-plugin-react)
Created by: [`jsx-eslint`](https://github.com/jsx-eslint)

## 📑 Recommended Read

- See how to use [flat config in ESLint 9](https://dev.to/aolyang/eslint-9-flat-config-tutorial-2bm5) 
- A tutorial for [creating your own ESLint rule](https://dev.to/meteor/how-to-create-your-own-eslint-rule-with-tests-boosting-the-dx-and-code-review-25el)
- A [migration guide to v9](https://medium.com/ekino-france/migrate-to-eslint-9-x-29727f790249?source=tag_recommended_feed---------1-85----------eslint----------7ded972d_2943_4352_9536_de642515ee72-------)
- [Understand the ESLint configurations](https://medium.com/@ritz.sh/understanding-eslint-configuration-eslintrc-js-vs-eslintrc-vs-eslintrc-json-287ec5e95bf4)

## 😎 Fun

![meme1](https://github.com/user-attachments/assets/29e07f56-5512-4a2d-ba03-a62d1ab0dcec)

---

Thank you for reading this month's newsletter; we look forward to your continued journey towards cleaner code and better linting practices.

**Reply by Jay-Karia** [reply]
I am not quite satisfied with the content of the "Version Updates" section,
here's what node.js did: https://nodejs.org/en/blog/weekly/weekly-update.2017-02-10 (few years ago).

I would like to hear your thoughts

**Reply by nzakas** [reply]
Are you just copying the highlights from the releases?

**Reply by Jay-Karia** [reply]
Yea, and what should be the optimal format for the releases section ?

**Reply by nzakas** [reply]
I think bullets are fine, but we probably want to group them in some way. I think rules changes should probably be grouped together, for example. For major releases we often describe things in terms of target audience: Users, Integrators, and Developers. It might be worth thinking about doing that here, too.

**Reply by Jay-Karia** [reply]
For the "version updates" section, I would like to have volunteer to write the content about the section, as I would not be able to get updated with versions of the eslint (what's new and all the stuff), as I would not be frequent with GitHub for now.

@amareshsm Are you willing to write the content for "version updates" section?

**Reply by Jay-Karia** [reply]
@nzakas your thoughts on other sections.

**Reply by nzakas** [reply]
Sorry, not sure what it is you're asking.

**Reply by Jay-Karia** [reply]
> Sorry, not sure what it is you're asking.

About the other sections in the format written above.

Other than the "version updates" section.

**Reply by Jay-Karia** [reply]
And i would not be able to write the content on the "version updates" section as I am not currently active on GitHub.

**Reply by nzakas** [reply]
The other sections look good to me.

I'm not sure I understand about "version updates". I'm assuming you'd just look that up on the blog when you write each newsletter?

**Reply by Jay-Karia** [reply]
Updated the "version updates" section and modified the newsletter a bit.
Is there anything "creative" or "unique" we can do to the newsletter ?

**Reply by nzakas** [reply]
That's up to you and @amareshsm . :smile: I really don't have the time to be actively involved.
